src/tools/spin/spinquic.cpp

```diff
@@ -243,7 +243,14 @@ struct SpinQuicGlobals {
 #ifndef FUZZING
             DumpMsQuicPerfCounters(MsQuic);
 #endif
+
+#ifdef QUIC_BUILD_STATIC
```

Comment:
Not an expert on static / dynamic linking, but a quick search says that for linux, dynamic linking also share global variables.
Should we simply always use the lock?

A comment to explain this would be useful, this isn't an obvious pattern IMO.

Reply:
You beat me to it. I was reasoning about this part too as it didn't make sense to me.

Reply:
@nibanks do you have any context? It seems wrong but I feel like changes like this were added for a reason.

Reply:
As I mentioned in the other comments, this was done on purpose. Dynamic linking MUST support parallel callers without synchronization from the callers.

Reply:
Can you explain why? I think we are all confused here about how it would be different between static and dynamic linking.